sound/soc/sof/topology.c

```diff
@@ -2128,9 +2131,19 @@ static int sof_widget_unload(struct snd_soc_component *scomp,
 		}
 		break;
 	case snd_soc_dapm_scheduler:
+		pipeline = swidget->private;
+
+		/*
+		 * For core 0, we will power down it at PM ops.
+		 * For other cores, if it is already powered off, just return.
+		 * TODO: add ref counts for each core, only power off it when
+		 *	 ref counts is decreased by 0.
+		 */
+		if (!(sdev->enabled_cores_mask & (1 << pipeline->core)) ||
```

Comment:
Do you need locking here around mask ? what happens if a pipeline is started when another topology is being loaded ?

Reply:
@lgirdwood that's good point, we might need locking at both load and unload, though we never had that yet.

Reply:
@lgirdwood comments addressed, please help review.